src/modules/ModuleTradeShared/composable.pair-input.ts

```diff
@@ -24,6 +25,42 @@ export function useLocalStorageAddrsOrigin(key: string, isActive?: Ref<boolean>)
   return offable
 }
 
+export function useRouteAddrsOrigin({
+  baseToken,
+  additionalBaseToken,
+  isActive,
+}: {
+  baseToken?: Address
+  additionalBaseToken?: Address
+  isActive?: Ref<boolean>
+}) {
+  const params = {
+    tokenA: useRouteParams('tokenA'),
+    tokenB: useRouteParams('tokenB'),
+  }
+
+  return computed({
+    get: () => {
+      if (!isActive?.value) return emptyPair()
+
+      const tokenA =
+        (typeof params.tokenA.value === 'string' && isAddress(params.tokenA.value) && params.tokenA.value) || null
+      const tokenB =
+        (typeof params.tokenB.value === 'string' && isAddress(params.tokenB.value) && params.tokenB.value) || null
+
+      if (tokenA && tokenB) return { tokenA, tokenB }
+      else if (tokenA && baseToken && tokenA !== baseToken) return { tokenA, tokenB: baseToken }
+      else if (tokenA && additionalBaseToken) return { tokenA, tokenB: additionalBaseToken }
```

Comment:
```suggestion
      if (tokenA && tokenB) return { tokenA, tokenB }
      if (tokenA && baseToken && !areAddressesEqual(tokenA, baseToken)) return { tokenA, tokenB: baseToken }
      if (tokenA && additionalBaseToken) return { tokenA, tokenB: additionalBaseToken }
```

It is critical to always compare addresses through `areAddressesEqual`.